src/app/workflow/view/view.component.html

```diff
@@ -40,3 +29,28 @@
     Delete
   </button>
 </ng-container>
+<button
+  type="button"
+  mat-button
+  color="accent"
+  class="btn-block"
+  *ngIf="!version.frozen && version.referenceType !== 'BRANCH'"
+  (click)="snapshotVersion()"
+  data-cy="dockstore-snapshot"
+  [matTooltip]="snapshotTooltip"
+  [disabled]="snapshotRequested"
+>
+  Snapshot
+</button>
+<button
+  mat-button
+  id="DOIButton"
+  color="accent"
+  class="btn-block"
+  (click)="requestDOIForWorkflowVersion()"
+  [disabled]="doiRequested"
```

Comment:
I think simply disabling both of these with isRefreshing$ would've been sufficient and would've avoided all the other problems

:man_shrugging:

Reply:
My bad then. When I pulled the code, I didn't even see a isRefreshing$ -- IDE as flagging it as a compile error.